### PR TITLE
fix: handling checkbox type, fix: proper value for sl-select, fix: prevent including sl-select value if empty, fix: prevent including sl-radio-group value if empty, fix: proper value if name attribute for sl-checkbox and sl-switch provided

### PR DESCRIPTION
**Pull request by karolskolasinski**
check out this topic: https://github.com/shoelace-style/shoelace/issues/744#event-11302002828

### CODE REVIEW

**Comment by karolskolasinski** [comment]
@benopotamus

**Comment by benopotamus** [comment]
@karolskolasinski Oops. Sorry I missed the notification for this PR. I'll give it a good look shortly.

File: htmx.ext.shoelace.js [hunk 1]
```diff
@@ -1,41 +1,49 @@
...
 
 /* Lightly modified version of the same function in htmx.js */
 function shouldInclude(elt) {
-
 	// sl-rating doesn't have a name attribute exposed through the Shoelace API
-	if (elt.tagName == 'SL-RATING' && elt.getAttribute('name')) {
+	if (elt.tagName === "SL-RATING" && elt.getAttribute("name")) {
 		return true;
 	}
-	if (elt.name === "" || elt.name == null || elt.disabled) {
```
Comment (benopotamus) [inline htmx.ext.shoelace.js:10]: Why did you remove the `elt.disabled` - did you find it was unnecessary?

Comment (karolskolasinski) [reply htmx.ext.shoelace.js:10]: yes, you are right:
Tip: Disabled <input> elements in a form will not be submitted! https://www.w3schools.com/tags/att_input_disabled.asp

Comment (karolskolasinski) [reply htmx.ext.shoelace.js:10]: @benopotamus

**Comment by karolskolasinski** [comment]
@benopotamus can you merge?

**Comment by benopotamus** [comment]
@karolskolasinski Of course. I just need to test it first - and it hasn't been a high enough priority for me lately. I have started and hope to finish it soon. Thank you for your patience! I do appreciate it.

File: htmx.ext.shoelace.js [hunk 2]
```diff
@@ -1,41 +1,49 @@
...
+					switch (elt.tagName) {
+						case "SL-CHECKBOX":
+						case "SL-SWITCH":
+							evt.detail.parameters[elt.name] = elt.value ?? elt.checked;
+							break;
+						case "SL-RATING":
+							evt.detail.parameters[elt.getAttribute("name")] = elt.value;
+							break;
+						case "SL-SELECT":
+							evt.detail.parameters[elt.name] = [elt.value];
```
Comment (benopotamus) [inline htmx.ext.shoelace.js:39]: Why is this better for an `sl-select` than `evt.detail.parameters[elt.name] = elt.value;`?

**Comment by benopotamus** [comment]
@karolskolasinski Thanks for submitting the PR. I've done some testing. Comments below :smile: 

> fix: handling checkbox type

This still returns `true` instead of `on` for checkboxes without values. I think the problem is on line 33, or I confused myself when testing.

> fix: proper value for sl-select

I think the improvement is on line 39, is that right? See comment added to that line.

> fix: prevent including sl-select value if empty

Native `select` will submit an empty value if an empty option is selected - e.g. `<option></option>`. I'd like to emulate that behaviour in this extension as well.

> fix: prevent including sl-radio-group value if empty

Looks good :+1:

> fix: proper value if name attribute for sl-checkbox and sl-switch provided

Looks good :+1:

File: htmx.ext.shoelace.js [hunk 3]
```diff
@@ -1,41 +1,49 @@
...
-
-					} else if (elt.tagName == 'SL-RATING') {
-						evt.detail.parameters[elt.getAttribute('name')] = elt.value
-
-					} else {
-						evt.detail.parameters[elt.name] = elt.value
+					switch (elt.tagName) {
+						case "SL-CHECKBOX":
+						case "SL-SWITCH":
+							evt.detail.parameters[elt.name] = elt.value ?? elt.checked;
```
Comment (benopotamus) [inline htmx.ext.shoelace.js:33]: I think maybe this shouldn't have the `?? elt.checked` part. `value` returns `on` if checked with no value attribute set on the input element.

**Comment by benopotamus** [comment]
I think there's a bug in Shoelace where it returns "undefined" instead of "on" when an sl-checkbox has no value - as demonstrated by this codepen https://codepen.io/benopotamus/pen/poYXMvZ. I've asked on the Shoelace Discord if someone can double check that for me before I add a bug to the Shoelace repo.

**Comment by benopotamus** [comment]
@karolskolasinski I updated the `shouldInclude` function to include some changes made to the htmx.js function of the same name. I had also been making little tweaks to the source file on my machine based on your PR and it was easier just to push both changes - so I've done that now. I think this PR won't be relevant anymore (hopefully!). 

Thank you for taking the time to contribute. Please create Github issues or another PR if you think there's more improvements to be made.

Also, the update I pushed resolves sl-checkboxes returning `'undefined'` instead of `'on'` when they don't have a value attribute set. I _think_ it's a Shoelace bug but that surprises me. We'll find out soon as I added a bug report for it to the Shoelace repo https://github.com/shoelace-style/shoelace/issues/1891..
